src/Novu.php

```diff
@@ -55,28 +56,37 @@ class Novu
     /**
      * Create a new Novu instance.
      *
-     * @param  string|null  $apiKey
+     * @param  array|string|null  $apiKey
      * @param  \GuzzleHttp\Client|null  $guzzle
      * @return void
      */
-    public function __construct($apiKey = null, HttpClient $client = null)
+    public function __construct($config = [], HttpClient $client = null)
     {
-        if ( is_null($apiKey)) {
-            throw IsNull::make('API KEY');
-        }
-
-        if( empty($apiKey)) {
-            throw isEmpty::make('API KEY');
+        // Default values
+        $defaultBaseUri = 'https://api.novu.co/v1/';
+    
+        if (is_string($config)) {
+            $apiKey = $config;
+            $baseUri = $defaultBaseUri;
+        } elseif (is_array($config)) {
+            $apiKey = $config['apiKey'] ?? null;
+            $baseUri = $config['baseUri'] ?? $defaultBaseUri;
+        } else {
+            throw new InvalidArgumentException("Invalid configuration provided.");
         }
-
-        if (! is_null($apiKey)) {
-            $this->setApiKey($apiKey, $client);
+    
+        if (is_null($apiKey)) {
+            throw IsNull::make('API KEY');
         }
-
-        if (! is_null($client)) {
-            $this->client = $client;
```

Comment:
Please can you add this back. This was added for folks to pass in their own client if they wish so. @isaiahdahl

Reply:
@unicodeveloper I understand that there is the ability for folks to use their own clients but I don't understand why those lines are needed back. Is there something I'm missing?  The `setApiKey` function on 98 will set `$this->client` because `$client` is passed in there.

Reply:
On a second note, you are right. We don't need it any longer because of the way it is set up there. @isaiahdahl